### PR TITLE
docs: introduce about webpack and metro bundler

### PR DESCRIPTION
**Pull request by bohdanprog**
# Summary
Short explanation on how to configure the webpack bundler and a brief rationale on why it's preferable to use the Metro bundler.

## Checklist
- [x] I added documentation in `README.md`

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -146,6 +146,62 @@ export default () => {
...
+>  ...
+>};
+>
+>module.exports = {
+>  ...config,
+>  module: {
+>      rules: [babelLoaderConfiguration],
+>    },
+>};
+> ```
```
Comment (jakex7) [inline USAGE.md:201]: Why is it quote?

File: USAGE.md [hunk 2]
```diff
@@ -146,6 +146,62 @@ export default () => {
...
+1. **Unified Development Experience**: Metro is the default bundler for React Native, providing a consistent experience across iOS, Android, and web platforms. This eliminates the need for managing separate build configurations for different platforms.
+
+2. **Optimized Performance**: Metro is designed specifically for React Native, offering faster build times and optimized performance, especially when dealing with large codebases.
+
+3. **Seamless Integration**: Expo offers out-of-the-box integration with Metro, simplifying the development process and reducing the need for complex configuration setups, which can be particularly beneficial when targeting multiple platforms.
+
+4. **Active Maintenance**: Unlike Webpack, Metro is actively maintained and continuously improved to support the latest React Native features. This ensures better support and access to new functionalities as they are released.
+
+5. **Simplified Configuration**: With Metro and Expo, there’s less overhead in terms of configuration compared to Webpack. This can make the setup process faster and less error-prone, especially for developers who are not as familiar with Webpack.
+
```
Comment (jakex7) [inline USAGE.md:165]: Do we need this section at all? It seems to be completely irrelevant to SVG library

File: USAGE.md [hunk 3]
```diff
@@ -146,6 +146,62 @@ export default () => {
...
+## Webpack bundler
+> [!WARNING] `expo/webpack-config`
+> 
+> `@expo/webpack-config` is deprecated and will not receive any new feature updates.
+
+> [!TIP] Migrate from Expo Webpack
+> 
+> If you are still using the Webpack bundler, we recommend migrating to Metro. This [documentation](https://docs.expo.dev/router/migrate/from-expo-webpack/) provides a good guide on how to migrate a website from Expo Webpack to Expo Metro.
```
Comment (jakex7) [inline USAGE.md:173]: Quote?
image

File: USAGE.md [hunk 4]
```diff
@@ -146,6 +146,62 @@ export default () => {
 };
 ```
 
+## Metro bundler
```
Comment (jakex7) [inline USAGE.md:149]: `##` makes it sub-header to 
> Use with content loaded from uri

Comment (jakex7) [reply USAGE.md:149]: We could probably make a `# Web configuration` header and a
`## Metro bundler` + `## Webpack` subheaders

File: USAGE.md (hunk 3 above)
Comment (jakex7) [reply USAGE.md:173]: Also, is it needed at all?

File: USAGE.md [hunk 5]
```diff
@@ -146,6 +146,62 @@ export default () => {
...
+You can set up your mobile and web applications using the Metro bundler, and you shouldn't face any issues using our library.
```
Comment (jakex7) [inline USAGE.md:150]:
```suggestion
No additional steps are required when using Metro bundler.
```

File: USAGE.md [hunk 6]
```diff
@@ -146,6 +146,62 @@ export default () => {
...
+## Webpack bundler
+> [!WARNING] `expo/webpack-config`
+> 
+> `@expo/webpack-config` is deprecated and will not receive any new feature updates.
+
+> [!TIP] Migrate from Expo Webpack
+> 
+> If you are still using the Webpack bundler, we recommend migrating to Metro. This [documentation](https://docs.expo.dev/router/migrate/from-expo-webpack/) provides a good guide on how to migrate a website from Expo Webpack to Expo Metro.
+
+If, for some reason, you are using the Webpack bundler, the following steps are important:
```
Comment (jakex7) [inline USAGE.md:175]:
```suggestion
If you are using the Webpack bundler, the following steps are needed:
```

**Review by jakex7 — CHANGES_REQUESTED** [review]

File: USAGE.md (hunk 2 above)
Comment (bohdanprog) [reply USAGE.md:165]: I added a short motivation on why using Metro instead of Webpack is better. We can remove that if you think it doesn't make any sense.

File: USAGE.md (hunk 3 above)
Comment (bohdanprog) [reply USAGE.md:173]: Why quote? I want to highlight that part.
In my opinion, it has to be here. However, I don't object if you believe it should be removed. I attempted to clarify it for the user and consolidate some brief tips in one location.

File: USAGE.md (hunk 1 above)
Comment (bohdanprog) [reply USAGE.md:201]: I want to highlight that part.

File: USAGE.md (hunk 2 above)
Comment (jakex7) [reply USAGE.md:166]: In that case, let's remove it completely

File: USAGE.md (hunk 1 above)
Comment (jakex7) [reply USAGE.md:202]: It's not highlighted, somehow Alerts are not rendered properly so it's displayed as a normal quote which makes it **less** visible
image

Comment (jakex7) [reply USAGE.md:202]: Also, don't put code snippets in quote blocks, please

File: USAGE.md (hunk 3 above)
Comment (jakex7) [reply USAGE.md:174]: Same as here https://github.com/software-mansion/react-native-svg/pull/2434/files#r1741596785
Also, this is not relevant to react-native-svg, so I don't see the point of posting it here.

File: USAGE.md (hunk 1 above)
Comment (bohdanprog) [reply USAGE.md:202]: something like this it's better?
Screenshot 2024-09-03 at 10 07 46

File: USAGE.md (hunk 3 above)
Comment (bohdanprog) [reply USAGE.md]: remoed.

File: USAGE.md (hunk 2 above)
Comment (bohdanprog) [reply USAGE.md]: removed.

File: USAGE.md [hunk 7]
```diff
@@ -145,6 +145,40 @@ export default () => {
...
+No additional steps are required when using Metro bundler.
+
+## Webpack
+If you are using the Webpack bundler, the following steps are needed:
+
+- Install `@react-native/assets-registry/registry` as a project dependency.
+- Ensure that you have the proper configuration to parse flow files from `@react-native/assets-registry/registry`.
+- Configure the Webpack `module -> rules` section and include an important rule for `node_modules/@react-native/assets-registry/registry`.
+
+#### Webpack Configuration
```
Comment (jakex7) [inline USAGE.md:160]: Useless header, you can add the filename instead.
It's `babel.config.js` right? Then simply
```suggestion
babel.config.js
```

**Review by jakex7 — CHANGES_REQUESTED** [review]

File: USAGE.md (hunk 7 above)
Comment (bohdanprog) [reply USAGE.md]: it's a webpack.config.js, I changed. that, thank you

**Review by jakex7 — APPROVED** [review]
LGTM
